packages/mantine/src/components/table/layouts/RowLayout.tsx

```diff
@@ -108,11 +114,12 @@ const RowLayoutBody = <T,>({table, doubleClickAction, getExpandChildren, loading
     const rows = table.getRowModel().rows.map((row) => {
         const rowChildren = getExpandChildren?.(row.original) ?? null;
         const isSelected = !!row.getIsSelected();
+        const shouldForceRowSelect = forceRowSelected && row.getIsSelected();
         const onClick = (event: MouseEvent<HTMLTableRowElement>) => {
             if (rowChildren) {
                 toggleCollapsible(event.currentTarget);
             }
-            if (!disableRowSelection && !multiRowSelectionEnabled) {
+            if (!disableRowSelection && !multiRowSelectionEnabled && !shouldForceRowSelect) {
```

Comment:
Won't that prevent use from selecting rows that aren't the initially selected one?

Reply:
So my thought process was that if the prop is passed to the layout and we click on a selected row, it won't toggle. But if the row isn't already selected, it will toggle and thus become selected.

Here's an example of how it's currently working in my ongoing admin-ui task (it's a bit hard to notice the clicks through a screen capture so when the text becomes highlighted it's actually me spamming the click on the row).

https://github.com/coveo/plasma/assets/133259861/0c9cb784-1194-4735-af07-d653439507e6